spec/jekyll-linkpreview_spec.rb

```diff
@@ -667,18 +684,61 @@ def check_default_template_without_image_is_rendered(html)
         end
 
         it 'cannot render custom template' do
-          html = @tag.render Liquid::Context.new
+          html = @tag.render @context
           check_default_template_without_image_is_rendered html
         end
       end
 
       context 'when no custom template file exists' do
         it 'cannot render custom template' do
-          html = @tag.render Liquid::Context.new
+          html = @tag.render @context
           check_default_template_without_image_is_rendered html
         end
       end
     end
+
+    context "when 'source' config value is modified" do
```

Comment:
Instead of repeating similar test cases for a modified source dir, maybe we can use [rspec-parameterized](https://rubygems.org/gems/rspec-parameterized) to make each test case DRY. I mean we can run a test case with each context.

Reply:
I added the rspec-parameterized to the gemspec and added three cases of site source locations.  
The entire `LinkPreviewTag#render` tests are now enclosed in `where do ~ end` and `with_them do ~ end`, enabling testing while being DRY as much as possible.   
I also enclosed the part where the templates were written for readability reasons.

The test cases are still confirmed to be passing.  

(I had trouble understanding the correct usage and had to commit more than I needed to. If you want me to squash the commits, please tell me.)